### PR TITLE
feat!: support for core 18.2

### PR DESCRIPTION
**Pull request by shuplenkov**
## Issue being fixed or feature implemented

Current version in incompatible with the latest core

## What was done?

Introduces getWallet method which includes all wallet related methods

## How Has This Been Tested?

With tests

## Breaking Changes

No

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by shumkov — APPROVED** [review]
👍

**Comment by strophy** [comment]
Should we add `listwallets` and `listwalletdir` RPCs, or were these intentionally omitted?

Are there any docs associated with this repo? It is not immediately clear to me how `getWallet()` and `setWallet()` should be used, some experimentation and reading the test code is necessary.

File: lib/index.js [hunk 1]
```diff
@@ -379,18 +394,52 @@ function generateRPCMethods(constructor, apiCalls, rpc) {
...
 
-  constructor.prototype.apiCalls = apiCalls;
+  return methods;
+}
+
+function getWalletRPCMethods(constructor, apiCalls, rpc) {
+
+  return {
+    getWallet: function (walletName) {
+      const path = '/wallet' + walletName;
```
Comment (strophy) [inline lib/index.js:412]:
```suggestion
      const path = '/wallet/' + walletName;
```

**Comment by strophy** [comment]
Could we also add `getmempoolentry` so I can check for IS locks, or does this lib provide a different way of checking that?

**Review by antouhou — APPROVED** [review]

**Pull request merged**